runtime/src/test/asset_test.rs

```diff
@@ -1013,6 +1015,592 @@ fn update_identifiers() {
     });
 }
 
+#[test]
+fn add_extension_successfully() {
+    build_ext().execute_with(|| {
+        let (owner_signed, owner_did) = make_account(AccountKeyring::Dave.public()).unwrap();
+
+        // Expected token entry
+        let token = SecurityToken {
+            name: b"TEST".to_vec(),
+            owner_did,
+            total_supply: 1_000_000,
+            divisible: true,
+            asset_type: AssetType::default(),
+        };
+
+        let ticker = Ticker::from_slice(token.name.as_slice());
+        assert!(!<identity::DidRecords>::exists(
+            Identity::get_token_did(&ticker).unwrap()
+        ));
+        let identifier_value1 = b"ABC123";
+        let identifiers = vec![(IdentifierType::Cusip, identifier_value1.to_vec())];
+        assert_ok!(Asset::create_token(
+            owner_signed.clone(),
+            owner_did,
+            token.name.clone(),
+            ticker,
+            token.total_supply,
+            true,
+            token.asset_type.clone(),
+            identifiers.clone(),
+        ));
+        // A correct entry was added
+        assert_eq!(Asset::token_details(ticker), token);
+
+        // Add smart extension
+        let extension_name = b"PTM";
+        let extension_id = AccountKeyring::Bob.public();
+
+        let extension_details = SmartExtension {
+            extension_type: SmartExtensionType::TransferManager,
+            extension_name: extension_name.to_vec(),
+            extension_id: extension_id.clone(),
+            is_archive: false,
+        };
+
+        assert_ok!(Asset::add_extension(
+            owner_signed.clone(),
+            ticker,
+            extension_details,
+        ));
+
+        // verify the data within the runtime
+        assert_eq!(
+            (Asset::extension_details((ticker, extension_id))).extension_type,
```

Comment:
We could call `Asset::extension_details((ticker, extension_id))` just one, and store  its result.
If type support `PartialEq`, you could just call one `assert_eq!( extension_details, extension_details_returned)`.